src/services/permissions-checker.js

```diff
@@ -101,89 +187,103 @@ class PermissionsChecker {
-        ._isAggregationFromScope(aggregator, conditions, expectedConditionFilters);
-
-      // NOTICE: Find in a condition correspond to a scope condition or not
-      const isScopeCondition = (condition) => PermissionsChecker
-        ._isConditionFromScope(condition, expectedConditionFilters.conditions);
-
-      // NOTICE: Perform a travel to find the scope in filters
-      const scopeFound = await parseFilters(
-        permissionInfos.filters,
-        isScopeAggregation,
-        isScopeCondition,
-      );
+  static async _isScopeValid(permissionInfos, scope) {
+    const expectedConditionFilters = PermissionsChecker
+      ._computeConditionFiltersFromScope(permissionInfos.userId, scope);
+
+    // NOTICE: Find aggregated condition. filtredConditions represent an array
+    //         of conditions that were tagged based on if it is present in the
+    //         scope
+    const isScopeAggregation = (aggregator, conditions) => PermissionsChecker
+      ._isAggregationFromScope(aggregator, conditions, expectedConditionFilters);
+
+    // NOTICE: Find in a condition correspond to a scope condition or not
+    const isScopeCondition = (condition) => PermissionsChecker
+      ._isConditionFromScope(condition, expectedConditionFilters.conditions);
+
+    // NOTICE: Perform a travel to find the scope in filters
+    const scopeFound = await parseFilters(
+      permissionInfos.filters,
+      isScopeAggregation,
+      isScopeCondition,
+    );
+
+    // NOTICE: In the case of only one expected condition, server will still send an aggregator
+    //         which will not match the request. If one condition is found and is from scope
+    //         then the request is valid
+    const isValidSingleConditionScope = !!scopeFound
+    && expectedConditionFilters.conditions.length === 1;
+
+    const isSameScope = !!scopeFound
+    && scopeFound.aggregator === expectedConditionFilters.aggregator
+    && !!scopeFound.conditions
+    && scopeFound.conditions.length === expectedConditionFilters.conditions.length;
+
+    return isValidSingleConditionScope || isSameScope;
+  }
 
-      // NOTICE: In the case of only one expected condition, server will still send an aggregator
-      //         which will not match the request. If one condition is found and is from scope
-      //         then the request is valid
-      const isValidSingleConditionScope = !!scopeFound
-        && expectedConditionFilters.conditions.length === 1;
+  static async _isCollectionBrowseAllowed(collectionPermissions, permissionInfos, scope) {
+    const { browseEnabled } = collectionPermissions.collection;
+    const { userId } = permissionInfos;
 
-      const isSameScope = !!scopeFound
-        && scopeFound.aggregator === expectedConditionFilters.aggregator
-        && !!scopeFound.conditions
-        && scopeFound.conditions.length === expectedConditionFilters.conditions.length;
+    if ((Array.isArray(browseEnabled) && !browseEnabled.includes(parseInt(userId, 10)))
```

Comment:
You've created a function that does exactly that 😅 `_isPermissionAllowed` 🤔

Reply:
😄  OMG that's true thanks!!